src/sections/staking/sections/dashboard/components/StakingInputSection/Stake/Stake.tsx

```diff
@@ -41,67 +43,63 @@ export const Stake = ({
 
   const form = useForm<{ amount: string }>()
 
-  const processedVotes = useProcessedVotesIds()
+  const { data: votes } = useProcessedVotesIdsQuery()
 
-  const onSubmit = async (values: FormValues<typeof form>) => {
-    const amount = getFixedPointAmount(values.amount, 12).toString()
+  const getExtrinsic = useCallback(
+    (amountHuman: string) => {
+      const amount = scale(amountHuman, 12).toString()
+
+      const isStakePosition = positionId != null
+
+      return isStakePosition
+        ? votes &&
+          (votes.newProcessedVotesIds.length ||
+            votes?.oldProcessedVotesIds.length)
+          ? api.tx.utility.batchAll([
+              ...votes.oldProcessedVotesIds.map((id) =>
+                api.tx.democracy.removeVote(id),
+              ),
+              ...votes.newProcessedVotesIds.map(({ classId, id }) =>
+                api.tx.convictionVoting.removeVote(
+                  classId ? classId : null,
+                  id,
+                ),
+              ),
+              api.tx.staking.increaseStake(positionId, amount),
+            ])
+          : api.tx.staking.increaseStake(positionId, amount)
+        : api.tx.staking.stake(amount)
```

Comment:
Maybe it would be better to reduce nesting if if...else statements in this case, so the code is more readable:

```suggestion
if (!isStakePosition) {
  return api.tx.staking.stake(amount);
}

if (votes && (votes.newProcessedVotesIds.length || votes?.oldProcessedVotesIds.length)) {
  return api.tx.utility.batchAll([
    ...votes.oldProcessedVotesIds.map((id) => api.tx.democracy.removeVote(id)),
    ...votes.newProcessedVotesIds.map(({ classId, id }) =>
      api.tx.convictionVoting.removeVote(classId || null, id)
    ),
    api.tx.staking.increaseStake(positionId, amount),
  ]);
}

return api.tx.staking.increaseStake(positionId, amount);
```